src/Features/InverseSquareLighting.cpp

```diff
@@ -0,0 +1,90 @@
+#include "InverseSquareLighting.h"
+#include "Features/InverseSquareLighting/Common.h"
+#include "LightLimitFix.h"
+
+void InverseSquareLighting::DrawSettings()
+{
+	editor.DrawSettings();
+}
+
+void InverseSquareLighting::EarlyPrepass()
+{
+	editor.GatherLights();
+}
+
+void InverseSquareLighting::PostPostLoad()
+{
+	stl::detour_thunk<CreatePointLight>(REL::RelocationID(17208, 17610));
+
+	logger::info("[InverseSquareLighting] Installed hooks");
+}
+
+RE::NiPointLight* InverseSquareLighting::CreatePointLight::thunk(RE::TESObjectLIGH* ligh, RE::TESObjectREFR* refr, RE::NiAVObject* root, bool forceDynamic, bool useLightRadius, bool affectRequesterOnly)
+{
+	const auto niLight = func(ligh, refr, root, forceDynamic, useLightRadius, affectRequesterOnly);
+
+	if (ligh && root && niLight)
+		SetExtLightData(niLight, ligh);
+
+	return niLight;
+}
+
+void InverseSquareLighting::SetExtLightData(RE::NiLight* niLight, const RE::TESObjectLIGH* ligh)
+{
+	const auto runtimeData = ISLCommon::RuntimeLightDataExt::Get(niLight);
+	runtimeData->flags.set(LightLimitFix::LightFlags::Initialised);
+	if (ligh->data.flags.any(static_cast<RE::TES_LIGHT_FLAGS>(ISLCommon::TES_LIGHT_FLAGS_EXT::kInverseSquare)))
+		runtimeData->flags.set(LightLimitFix::LightFlags::InverseSquare);
+	runtimeData->cutoffOverride = std::clamp(ligh->data.fallofExponent, 0.01f, 1.f);
+	runtimeData->lighFormId = ligh->formID;
+}
+
+void InverseSquareLighting::ProcessLight(LightLimitFix::LightData& light, RE::BSLight* bsLight, RE::NiLight* niLight) const
+{
+	const auto runtimeData = ISLCommon::RuntimeLightDataExt::Get(niLight);
+
+	if (light.lightFlags.none(LightLimitFix::LightFlags::Initialised)) {
+		const auto userData = niLight->GetUserData();
+		logger::debug("[InverseSquareLighting] FormID: 0x{:08X} | Light*: {:p} | Name: {} - light uninitialised", userData ? userData->formID : 0, static_cast<void*>(niLight), niLight->name);
+		runtimeData->flags.set(LightLimitFix::LightFlags::Initialised);
+	}
+
+	const bool isInvSq = light.lightFlags.any(LightLimitFix::LightFlags::InverseSquare);
+	if (bsLight->pointLight && editor.enabled && ((isInvSq && editor.disableInvSqLights) || (!isInvSq && editor.disableRegularLights)))
```

Comment:
Do we need to check bsLight for nullptr?  Consider across any pointer checks.

Reply:
In both calling functions leading here the BSLight is already being accessed so can be assumed to be valid